services/automerge/automerge.go

```diff
@@ -111,6 +104,33 @@ func MergeScheduledPullRequest(ctx context.Context, sha string, repo *repo_model
 	return nil
 }
 
+// StartPRCheckAndAutoMerge start an automerge check and auto merge task for a pull request
+func StartPRCheckAndAutoMerge(ctx context.Context, pull *issues_model.PullRequest) {
```

Comment:
Why not call `StartPRCheckAndAutoMergeBySHA` to avoid duplicate code?

Reply:
They are two different functions. I think less code are duplicated.

Reply:
Actually I don't see why `sha` should be passed into `addToQueue`

It seems that the `sha` is only used for logging. Ideally only call `addToQueue(pr.ID)` should be enough, and a lot of code could be simplified?

Reply:
I think the SHA will prevent when both an automerge task and a newly push are triggered. So the previous triggered automerge task will be ignored.
```
if headCommitID != sha {
		log.Warn("Head commit id of auto merge %-v does not match sha [%s], it may means the head branch has been updated. Just ignore this request because a new request expected in the queue", pr, sha)
		return
	}
```

Reply:
Actually such check doesn't seem right and only makes some problem harder to debug.

Because, the head commit ID could change again after that check.